### PR TITLE
add dbCommit

### PR DESCRIPTION
**Pull request by mingzhenliu**

### CODE REVIEW

File: fisco-bcos/blockverifier/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+target_link_libraries(test_verifier ethcore)
+target_link_libraries(test_verifier p2p)
+target_link_libraries(test_verifier blockverifier)
+target_link_libraries(test_verifier executivecontext)
+target_link_libraries(test_verifier evm)
+target_link_libraries(test_verifier sync)
+target_link_libraries(test_verifier txpool)
+target_link_libraries(test_verifier mptstate)
+target_link_libraries(test_verifier blockchain)
+target_link_libraries(test_verifier consensus)
```
Comment (wheatli) [inline fisco-bcos/blockverifier/CMakeLists.txt]: why would this verifier depend on consensus?

File: fisco-bcos/blockverifier/verifierMain.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+ * verifierMain.cpp
+ *
+ *  Created on: 2018年10月26日
```
Comment (wheatli) [inline fisco-bcos/blockverifier/verifierMain.cpp]: why there are chinese???

File: libblockverifier/BlockVerifier.cpp [hunk 3]
```diff
@@ -33,18 +33,22 @@ using namespace dev::executive;
...
 ExecutiveContext::Ptr BlockVerifier::executeBlock(Block& block)
 {
-    LOG(TRACE) << "BlockVerifier::executeBlock tx_num=" << block.transactions().size();
+    LOG(TRACE) << "BlockVerifier::executeBlock tx_num=" << block.transactions().size()
+               << " hash: " << block.blockHeader().hash()
+               << " num: " << block.blockHeader().number()
+               << " stateRoot: " << block.blockHeader().stateRoot();
     ExecutiveContext::Ptr executiveContext = std::make_shared<ExecutiveContext>();
-    try
+    // try
```
Comment (wheatli) [inline libblockverifier/BlockVerifier.cpp]: Just delete the brace {}  when you drop try/catch logic

File: libblockverifier/ExecutiveContext.cpp [hunk 4]
```diff
@@ -131,4 +131,10 @@ void ExecutiveContext::setPrecompiledContract(
...
 {
     m_precompiledContract = precompiledContract;
-}
\ No newline at end of file
+}
+
+void ExecutiveContext::dbCommit()
+{
+    m_stateFace->dbCommit(m_blockInfo.hash, m_blockInfo.number.convert_to<int64_t>());
+    m_memoryTableFactory->commitDB(m_blockInfo.hash, m_blockInfo.number.convert_to<int64_t>());
```
Comment (wheatli) [inline libblockverifier/ExecutiveContext.cpp]: what will happen if the dbCommit success but commitDB failed ? I think we should use logic to   handle this situation

File: libexecutive/Executive.cpp [hunk 5]
```diff
@@ -95,7 +79,7 @@ void Executive::initialize(Transaction const& _transaction)
         // Avoid unaffordable transactions.
         bigint gasCost = (bigint)m_t.gas() * m_t.gasPrice();
         bigint totalCost = m_t.value() + gasCost;
-        if (m_s.balance(sender) < totalCost)
+        /*if (m_s.balance(sender) < totalCost)
```
Comment (wheatli) [inline libexecutive/Executive.cpp]: Would you like to give some comments for why you comment this logic

File: test/unittests/libblockverifier/BlockVerifierTest.cpp [hunk 6]
```diff
@@ -106,7 +114,36 @@ BOOST_FIXTURE_TEST_SUITE(BlockVerifierTest, BlockVerifierFixture);
 
 BOOST_AUTO_TEST_CASE(executeBlock)
 {
-    // m_blockVerifier->executeBlock(m_fakeBlock->getBlock());
+    /* for (int i = 0; i < 100; ++i)
```
Comment (wheatli) [inline test/unittests/libblockverifier/BlockVerifierTest.cpp]: why ?

File: libblockverifier/ExecutiveContext.cpp (hunk 4 above)
Comment (mingzhenliu) [reply libblockverifier/ExecutiveContext.cpp]: m_memoryTableFactory->commitDB is really written.

File: libexecutive/Executive.cpp (hunk 5 above)
Comment (mingzhenliu) [reply libexecutive/Executive.cpp]: fisco_bcos does not have the concept of account balance, so there is no gas consumption.

File: fisco-bcos/blockverifier/CMakeLists.txt (hunk 1 above)
Comment (mingzhenliu) [reply fisco-bcos/blockverifier/CMakeLists.txt]: It's only convenient to expand the demo later.

File: test/unittests/libblockverifier/BlockVerifierTest.cpp (hunk 6 above)
Comment (mingzhenliu) [reply test/unittests/libblockverifier/BlockVerifierTest.cpp]: It needs to be improved and commented first.

File: fisco-bcos/blockverifier/CMakeLists.txt (hunk 1 above)
Comment (wheatli) [reply fisco-bcos/blockverifier/CMakeLists.txt]: It's not allowed~

File: libblockverifier/ExecutiveContext.cpp (hunk 4 above)
Comment (wheatli) [reply libblockverifier/ExecutiveContext.cpp]: you mean only the 'm_memoryTableFactory->commitDB‘ will write data into disk ?

File: libexecutive/Executive.cpp (hunk 5 above)
Comment (wheatli) [reply libexecutive/Executive.cpp]: If so, Just delete these useless codes

File: libblockverifier/ExecutiveContext.cpp (hunk 4 above)
Comment (mingzhenliu) [reply libblockverifier/ExecutiveContext.cpp]: yes

File: libexecutive/Executive.cpp (hunk 5 above)
Comment (mingzhenliu) [reply libexecutive/Executive.cpp]: ok

**Pull request merged**